internal/k8s/reconciler/route_test.go

```diff
@@ -173,8 +173,6 @@ func TestRouteValidate(t *testing.T) {
 		Reference: &service.BackendReference{},
 	}
 
-	resolver.EXPECT().Resolve(gomock.Any(), reference).Return(resolved, nil)
-
 	route := NewK8sRoute(&gwv1alpha2.HTTPRoute{
```

Comment:
Not sure if I missed some stuff like this in the initial PR or if it's not that big of a deal because it eventually gets dropped anyway -- but thinking we should drop the public initializer functions in this PR if we're introducing the factories.

Reply:
@andrewstucki Thoughts on just making them private instead of public? In particular, `NewK8sGateway` has a good bit of "special" init logic for pulling listeners off of the kubernetes `Gateway` object. It feels better to me to have that live in `gateway.go` for example rather than `factory.go`.

Reply:
I'm fine with that for now, especially as this code is very much "in-flux" until we're closer to the full refactor merge, at which point we can deal with changing this up if necessary.

Reply:
Un-exported `NewK8sRoute` and `NewK8sGateway` 💯